File: 코딩테스트 고득점 Kit/탐욕법/체육복.py
def solution(n, lost, reserve):
    answer = 0
    arr = [1]*n
    
    for l in lost: # 두고온 넘들
        arr[l-1] -= 1
    for r in reserve:
        arr[r-1] += 1
    
    for i in range(len(arr)):
        if arr[i] == 0: # 체육복이 없다면
            if i > 0 and arr[i-1] >= 2:
                arr[i-1] -= 1
                arr[i] += 1
                
            elif i < n-1 and arr[i+1] >= 2: # 여분이 있다면
                    arr[i+1] -= 1
                    arr[i] += 1
    
    for a in arr:
        if a > 0:
            answer += 1
    return answer

File: 코딩테스트 고득점 Kit/탐욕법/체육복/체육복.swift
import Foundation

import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var answer:Int = 0
    var students:[Int] = Array(repeating: 1, count: n+1)
    
    for r in reserve {
        students[r] += 1
    }
    
    for l in lost {
        students[l] -= 1
    }
    
    //print(students)
    
    for i in 1...n {
        if students[i] == 0 {
            if i > 1 && students[i-1] > 1{
                students[i-1] -= 1
                students[i] = 1
            } else if i < n && students[i+1] > 1 {
                students[i+1] -= 1
                students[i] = 1
            }
        }
    }
    
    //print(students)
    answer = students.filter { $0 >= 1}.count-1
    
    return answer
}